### ISSUE TITLE
Ardent Defender Not Working

### ISSUE THREAD

**Issue by assassinbait** [chromiecraft/chromiecraft#5852]
### What client do you play on?

enUS

### Faction

Alliance

### Content Phase:

Generic

### Current Behaviour

Ardent Defender does not negate the killing blow and heal the user, it simply does not proc.

#4522 In this ticket, someone said that it may be due to overkill damage, which made sense to me since i was running heroic dungeons. So i tested this to verify;

![b5pXprk](https://github.com/chromiecraft/chromiecraft/assets/84478740/36723c13-fa0d-4931-ad26-2fd18d32aae4)

That does not seem to be the case

### Expected Blizzlike Behaviour

The talent should negate the killing blow, regardless of the amount of damage taken, and heal the user up to 30% of their maximum health.

### Source

![image](https://github.com/chromiecraft/chromiecraft/assets/84478740/1415b955-058a-4e57-939b-e3a8f0548e83)

### Steps to reproduce the problem

find an enemy
attack it
die

### Extra Notes

this has only started happening today. the talent worked perviously. i have tried resetting talents, swapping specializations, restarting my game and the /reload command but to no avail.

It doesn't seem like something specific causes this to happen. The issue happened persistently in different locations and situations.

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/a779cca9e3a3d673761d6bbe6d1c5d1f51c2f6f3

### Operating system

Ubuntu 20.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-bg-item-reward](https://github.com/azerothcore/mod-bg-item-reward)
- [mod-BreakingNewsOverride](https://github.com/AnchyDev/BreakingNewsOverride)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-cta-switch](https://github.com/azerothcore/mod-cta-switch)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna](https://github.com/azerothcore/mod-eluna)
- [mod-instanced-worldbosses](https://github.com/nyeriah/mod-instanced-worldbosses)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-low-level-rbg](https://github.com/azerothcore/mod-low-level-rbg)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-progression-system](https://github.com/azerothcore/mod-progression-system)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-rdf-expansion](https://github.com/azerothcore/mod-rdf-expansion)
- [mod-transmog](https://github.com/azerothcore/mod-transmog)
- [mod-weekend-xp](https://github.com/azerothcore/mod-weekend-xp)
- [mod-zone-difficulty](https://github.com/azerothcore/mod-zone-difficulty)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-event-scripts](https://github.com/55Honey/Acore_eventScripts)
- [lua-exchange-npc](https://github.com/55Honey/Acore_ExchangeNpc)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)

### Customizations

None

### Server

ChromieCraft

**Comment by Cooties-cc** [comment]
[AD test.webm](https://github.com/chromiecraft/chromiecraft/assets/62584589/cd4a5c53-36d4-4751-b124-c25912fb0715)
Seems to be working fine to me on PTR. I would suggest maybe including combat logs to show it not working. 
It might also just be a personal issue, which would need a ticket in discord (something like the 2min debuff not going away but also not showing, perhaps).

**Comment by balleny** [comment]
a small correction, you wrote
> The talent should negate the killing blow, regardless of the amount of damage taken, and heal the user up to 30% of their maximum health.

this isnt accurate. technically, it absorbs the hit that would kill you and instead restores you to x% of hp.
there are some effects that circumvent absorb effects (e.g. warlocks Chaosbolt) and those will not trigger Argent Defender but kill you instead.

**Comment by mussy44** [comment]
> [AD test.webm](https://github.com/chromiecraft/chromiecraft/assets/62584589/cd4a5c53-36d4-4751-b124-c25912fb0715) Seems to be working fine to me on PTR. I would suggest maybe including combat logs to show it not working. It might also just be a personal issue, which would need a ticket in discord (something like the 2min debuff not going away but also not showing, perhaps).

it does seem to be a personal issue. i have seen the effect proc on other people yesterday.

i went to discord and instead was told to make a report here. i will post on discord again i guess.

> a small correction, you wrote
> 
> > The talent should negate the killing blow, regardless of the amount of damage taken, and heal the user up to 30% of their maximum health.
> 
> this isnt accurate. technically, it absorbs the hit that would kill you and instead restores you to x% of hp. there are some effects that circumvent absorb effects (e.g. warlocks Chaosbolt) and those will not trigger Argent Defender but kill you instead.

right, that's good to know. is there a list of spells that ignore the absorb effect?

**Comment by mussy44** [comment]
seems to be cause of dualspec. did a couple of tests and ardent defender wouldnt proc after switching from my offspec to my main spec, and it only got fixed after i relogged and reset talents

**Comment by Vyperhand** [comment]
![image](https://github.com/chromiecraft/chromiecraft/assets/131016474/fb36e830-a699-42b1-bb3e-40186f19ff3d)

**Comment by Dotieee** [comment]
Yup, same here. If I swap specs using dual spec things just stop working and I die instantly as a tank. I have to go to trainer and manually respec to get it to work again.

**Comment by Depe79** [comment]
Redoubt bugs too when switching specs and i'm pretty sure one time happened on righteous fury.

**Comment by Salehyn** [comment]
Testing together with Choof today yielded these results:

Respecing to prot with dual spec while inside heroic TBC dungeon (with or without RDF) or inside TBC Raids such as kara causes the Ardent Defender effect to become inactive.
Specing to prot from a blank talent spec (no dual spec) while inside heroic TBC dungeon or TBC raids such as kara causes the Ardent Defender effect to become inactive.
Doing any of this in a normal dungeon, classic dungeon (lvl 60) or outside dungeon will not result in bug.

There seems to be a problem with the game applying the aura belonging to AD when inside dungeons that have been tuned by CC. Please look at the custom nerf auras interfering with this spell.

**Comment by Salehyn** [comment]
Further notes: there have been reported cases where the talent bugged without dungeons or raids involved, so it is possible there is another source of this bug as well.
Other talents such as redoubt still require testing to see if they suffer from the same bug.

**Comment by Nyeriah** [comment]
Did this happen during CC vanilla? How far back was the first occurrence?
